spec/controllers/request_controller_spec.rb

```diff
@@ -497,6 +501,27 @@ def get_attachment(params = {})
       get_attachment
     end
 
+    it 'sets the correct read permissions for the new file' do
+      # ensure the cache directory exists
+      get_attachment
+
+      param =
+        ActionController::Parameters.new(default_params.merge(only_path: true))
+      key_path = @controller.foi_fragment_cache_path(param)
+
+      # remove the pre-existing cached file
+      File.delete(key_path)
+
+      # create a new cache file and check the file permissions
+      get_attachment
+      octal_stat = sprintf("%o", File.stat(key_path).mode)[-3..-1]
+      if rails5?
+        expect(octal_stat).to eq('644')
+      else
+        expect(octal_stat).to match(/^6(4|6)4$/)
```

Comment:
Yeah, I could have added a helpful comment here. It boils down to "we only care about this for Rails 5 but writing a Rails 5 only test here would be weird"

In Rails 5, we start from `0600` as that's what `Tempfile.new` sets - the minimum permissions for the current user to read and write to a file that's going to be discarded. And we know exactly what we'll end up with as we're manually setting the rest.

In Rails 4, the permissions are "whatever the `umode` of the owning user is" (copied from the containing directory which we created before calling `atomic_write`) which for us is `0644` and for Travis appears to be `0664`. We've not felt the need to test this before as in production, it's "whatever defaults you set for the process owner", which seems fair.

I need to test this theory but I'm pretty sure we don't care what that middle digit is at all - we know the owner should have `rw` (from `Tempfile.new`), we know that 'others' should have `r` because we just set it, I don't think we need to care about 'group' for file access - if the group can write as well on Travis, fine!

Reply:
Theory checks out - have set one of the attachment files on the staging site to `0604` and it works just fine.

So it needs `0604` as a minimum to work with a web container. Running locally is misleading as even `0600` will work as the files get served up by the Rails server.